sigal/plugins/encrypt/encrypt.py

```diff
@@ -0,0 +1,275 @@
+# copyright (c) 2020 Bowen Ding
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+
+'''Plugin to protect gallery by encrypting image files using a password.
+
+Options:
+    encrypt_options = {
+        'password': 'password',
+        'ask_password': False,
+        'gcm_tag': 'randomly_generated',
+        'kdf_salt': 'randomly_generated',
+        'kdf_iters': 10000,
+        'encrypt_symlinked_originals': False
+    }
+
+- ``password``: The password used to encrypt the images on gallery build, 
+  and decrypt them when viewers access the gallery. No default value. You must
+  specify a password.
+- ``ask_password``: Whether or not viewers are asked for the password to view 
+  the gallery. If set to ``False``, the password will be present in the HTML files
+  so the images are decrypted automatically. Defaults to ``False``.
+- ``gcm_tag``, ``kdf_salt``, ``kdf_iters``: Cryptographic parameters used when
+  encrypting the files. ``gcm_tag``, ``kdf_salt`` are meant to be randomly generated, 
+  ``kdf_iters`` defaults to 10000. Do not specify them in the config file unless 
+  you have good reasons to do so.
+- ``encrypt_symlinked_originals``: Force encrypting original images even if they 
+  are symlinked. If you don't know what it means, leave it to ``False``.
```

Comment:
I think this should be True by default, and we could maybe even remove this setting (and send a warning that orig_link should not be used in this case). When using this plugin clearly the goal is to protect privacy by encrypting all images including the original ones.

Reply:
Yes, but we need to also make sure people don't get their original pictures overwritten if they accidentally have both `keep_orig` and `orig_link` set to `True`, and have not made a backup. I think it's better we abort the encryption and send a warning, to make sure the user understands the situation, then it's up to them to decide. It's trivial to restart an aborted gallery build, while to recover the originals by decrypting them is not.

Reply:
Agreed! So I think you can remove this setting ?

Reply:
Remove it from the doc or the code?

Reply:
Remove from the code, as you said above we should abort if `keep_orig` and `orig_link` are True, so I don't think we should keep this setting to force encryption. People should just choose between `orig_link` and using this plugin.

Reply:
Good point. Changed in a3b0876.